internal/driver/testdata/testflame.js

```diff
@@ -0,0 +1,90 @@
+function TestFlame() {
+  const test = new TestFixture();
+  const chart = document.getElementById("stack-chart");
+  if (!chart) {
+    test.err("could not find stack-chart");
+    return;
+  }
+  const chartRect = chart.getBoundingClientRect();
+
+  // Create map from box text to DOM element.
+  // TODO: Generalize to support multiple boxes for a given piece of text.
+  let boxMap;
+  fetchBoxes();
+  function fetchBoxes() {
+    boxMap = new Map();
+    const boxes = document.querySelectorAll(".boxbg");
+    for (let box of boxes) {
+      const text = box.innerText;
+      boxMap.set(text, box);
+    }
+  }
+
+  // rect gets the bounding box for box with text t.
+  function rect(t) {
+    const elem = boxMap.get(t);
+    if (!elem) {
+      test.err("did not find", t);
+      return null;
+    }
+    return elem.getBoundingClientRect();
+  }
+
+  // checkCalls checks that box with text a is positioned w.r.t. box with
+  // text b to indicate a call from a to b.  Expect a gap of the specified
+  // number of rows.
+  function checkCalls(a, b, gap = 0) {
+    test.setContext("checkCalls", a, b, gap);
+    const ra = rect(a);
+    const rb = rect(b);
+    if (!ra || !rb) return;
+
+    const pixelGap = gap * rb.height;
+    if (rb.top != ra.bottom + pixelGap) {
+      test.err("not above");
+    }
+    // TODO: Allow checking boxes above pivots.
+    if (rb.left < ra.left || rb.right > ra.right) {
+      test.err("horizontal span of", a, "is not nested inside horizontal span of", b);
+    }
+  }
+
+  // checkWidth checks that the width of the box with text t is approximately
+  // the specified fraction of the total width.
+  function checkWidth(t, fraction) {
+    test.setContext("checkWidth", t, fraction);
+    const r = rect(t);
+    if (!r) return;
+    const expect = (chartRect.width - 4) * fraction;
```

Comment:
nit: a named constant for the 4 might be good, it's a bit obscure as is.

Reply:
My 2 cents: personally I think amending the commit is better than squashing it into one, because it would be easier to relate the comments and CI runs to code in the PR.

Reply:
I think you meant "adding a new commit" instead of "amending the commit"?

I agree with the decision to not squash until merging into main. Also makes it easier for reviewers to see the changes since the last round of reviews.

Thanks.

Reply:
I don't have strong preference, separate commits are easier so probably the default choice. One added advantage is being able to move out a specific part of the change into a separate PR if requested by a reviewer - easier to do if it's a separate commit. Doesn't happen that often though, at least for this repo.